internal/user/user_profile_hash_v1.go

```diff
@@ -6,54 +6,55 @@ import (
 	"io"
 
 	"github.com/pkg/errors"
-	"golang.org/x/crypto/scrypt"
+
+	"github.com/kopia/kopia/internal/crypto"
 )
 
 // parameters for v1 hashing.
 const (
 	hashVersion1 = 1
 
-	v1ScryptN    = 65536
-	v1ScryptR    = 8
-	v1ScryptP    = 1
 	v1SaltLength = 32
-	v1KeyLength  = 32
 )
 
 //nolint:gochecknoglobals
-var dummyV1HashThatNeverMatchesAnyPassword = make([]byte, v1KeyLength+v1SaltLength)
+var dummyV1HashThatNeverMatchesAnyPassword = make([]byte, crypto.MasterKeyLength+v1SaltLength)
 
 func (p *Profile) setPasswordV1(password string) error {
 	salt := make([]byte, v1SaltLength)
 	if _, err := io.ReadFull(rand.Reader, salt); err != nil {
 		return errors.Wrap(err, "error generating salt")
 	}
 
+	var err error
 	p.PasswordHashVersion = 1
-	p.PasswordHash = computePasswordHashV1(password, salt)
+	p.PasswordHash, err = computePasswordHashV1(password, salt)
 
-	return nil
+	return err
 }
 
-func computePasswordHashV1(password string, salt []byte) []byte {
-	key, err := scrypt.Key([]byte(password), salt, v1ScryptN, v1ScryptR, v1ScryptP, v1KeyLength)
+func computePasswordHashV1(password string, salt []byte) ([]byte, error) {
+	key, err := crypto.DeriveKeyFromPassword(password, salt, crypto.DefaultKeyDerivationAlgorithm)
 	if err != nil {
-		panic("unexpected scrypt error")
+		return nil, err
 	}
 
 	payload := append(append([]byte(nil), salt...), key...)
 
-	return payload
+	return payload, nil
 }
 
 func isValidPasswordV1(password string, hashedPassword []byte) bool {
-	if len(hashedPassword) != v1SaltLength+v1KeyLength {
+	if len(hashedPassword) != v1SaltLength+crypto.MasterKeyLength {
 		return false
 	}
 
 	salt := hashedPassword[0:v1SaltLength]
 
-	h := computePasswordHashV1(password, salt)
+	h, err := computePasswordHashV1(password, salt)
+	if err != nil {
+		return false
```

Comment:
Returning a `false` is still a valid response from `isValidPasswordV1()`. It seems like this function has a tri-state, including error state.

Would be good to log the error from `computePasswordHashV1` as well

if you believe we should not hit the error ever then we should panic() instead.

Reply:
Hmm, it's very possible that we can hit an error here (lets say an invalid algorithm is passed). So would logging and retuning false be sufficient?

Reply:
I can `log.Fatal(err)` so that way we will quit with an error.

Reply:
`log.Fatal()` works, thanks!